### ISSUE TITLE
Uncaught TypeError: Cannot read property 'matchesParams' of undefined

### ISSUE THREAD

**Issue by bsr203** [atom/find-and-replace#855]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.14.0 x64
**Electron**: 1.3.13
**OS**: Mac OS X 10.9.5
**Thrown From**: [find-and-replace](https://github.com/atom/find-and-replace) package 0.206.1

### Stack Trace

Uncaught TypeError: Cannot read property 'matchesParams' of undefined

```
At /Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/marker-layer.js:191

TypeError: Cannot read property 'matchesParams' of undefined
  at /app.asar/node_modules/text-buffer/lib/marker-layer.js:191:22
  at Set.forEach (native)
  at MarkerLayer.module.exports.MarkerLayer.findMarkers (/app.asar/node_modules/text-buffer/lib/marker-layer.js:187:17)
  at DisplayLayer.destroyFoldsIntersectingBufferRange (/app.asar/node_modules/text-buffer/lib/display-layer.js:227:29)
  at TextEditor.module.exports.TextEditor.destroyFoldsIntersectingBufferRange (/app.asar/src/text-editor.js:3089:32)
  at Selection.module.exports.Selection.setBufferRange (/app.asar/src/selection.js:98:21)
  at /app.asar/src/text-editor.js:2169:29
  at TextEditor.module.exports.TextEditor.mergeSelections (/app.asar/src/text-editor.js:2521:43)
  at TextEditor.module.exports.TextEditor.mergeIntersectingSelections (/app.asar/src/text-editor.js:2487:35)
  at TextEditor.module.exports.TextEditor.setSelectedBufferRanges (/app.asar/src/text-editor.js:2162:19)
  at TextEditor.module.exports.TextEditor.setSelectedBufferRange (/app.asar/src/text-editor.js:2145:19)
  at TextEditor.module.exports.TextEditor.setSelectedScreenRange (/app.asar/src/text-editor.js:2194:19)
  at FindView.module.exports.FindView.selectMarkerAtIndex (/app.asar/node_modules/find-and-replace/lib/find-view.js:754:16)
  at FindView.module.exports.FindView.selectFirstMarkerAfterCursor (/app.asar/node_modules/find-and-replace/lib/find-view.js:620:19)
  at /app.asar/node_modules/find-and-replace/lib/find-view.js:3:59
  at FindView.module.exports.FindView.findAndSelectResult (/app.asar/node_modules/find-and-replace/lib/find-view.js:504:9)
  at FindView.findAndSelectResult (/app.asar/node_modules/find-and-replace/lib/find-view.js:3:59)
  at FindView.module.exports.FindView.findNext (/app.asar/node_modules/find-and-replace/lib/find-view.js:486:19)
  at FindView.findNext (/app.asar/node_modules/find-and-replace/lib/find-view.js:3:59)
  at FindView.module.exports.FindView.confirm (/app.asar/node_modules/find-and-replace/lib/find-view.js:441:19)
  at /app.asar/node_modules/find-and-replace/lib/find-view.js:285:26)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/app.asar/src/command-registry.js:259:29)
  at /app.asar/src/command-registry.js:3:59
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:599:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:390:22)
  at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/app.asar/src/window-event-handler.js:106:36)
  at /app.asar/src/window-event-handler.js:3:59)

```

### Commands

```
     -3:49.5.0 core:paste (input.hidden-input)
     -3:18.9.0 core:copy (input.hidden-input)
     -3:16.1.0 nuclide-file-tree:reveal-tab-file (div.title.icon.js-icon.medium-yellow)
     -3:16.1.0 nuclide-file-tree:toggle (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-isotope-ui.find-visible)
     -3:07.6.0 nuclide-file-tree:search-in-directory (span.)
     -3:07.6.0 project-find:show-in-current-directory (span.)
     -3:07.5.0 core:paste (input.hidden-input)
     -3:07.4.0 core:confirm (input.hidden-input)
     -2:54.3.0 find-and-replace:show (input.hidden-input)
  9x -2:53.8.0 core:confirm (input.hidden-input)
     -2:36 find-and-replace:show (input.hidden-input)
  7x -2:35.2.0 core:confirm (input.hidden-input)
     -0:18.7.0 find-and-replace:show (input.hidden-input)
  6x -0:17.7.0 core:confirm (input.hidden-input)
  3x -0:10.5.0 find-and-replace:show-previous (input.hidden-input)
  5x -0:07.0 core:confirm (input.hidden-input)
```

### Non-Core Packages

```
autocomplete-paths 1.0.5 
file-icons 2.0.15 
fold-comments 0.6.0 
go-debug 1.0.4 
go-plus 5.1.1 
highlight-selected 0.12.0 
isotope-ui 2.7.1 
language-babel 2.54.1 
lcov-info 0.12.0 
linter 1.11.21 
linter-eslint 8.1.0 
merge-conflicts 1.4.4 
nuclide 0.205.0 
nuclide-format-js 0.0.36 
prettier-atom 0.16.0 
project-manager 3.3.3 
sort-lines 0.14.0 
terminal-plus 0.14.5 
tool-bar 1.0.1 
```

**Comment by bsr203** [comment]
also see https://github.com/atom/atom/issues/6899

**Comment by nathansobo** [comment]
@bsr203 Any details about your usage pattern preceding this exception that might help with a reproduction? I'm installing all the same packages as you and giving it a try.

**Comment by bsr203** [comment]
@nathansobo I cannot reproduced it anymore. I upgraded atom and all packages in the morning, and restarted (most likely). I was getting this errors few times as I scroll through a javascript file with few lint warnings, or when do find a string in the same file. I wasn't getting this error when just opening this file or finding the pattern couple of times, but scrolling through it towards the end or finding the pattern few times. 

I restarted atom again, and it is not occurring anymore. you are free to close this as it might be because I didn't restart after upgrade properly.

**Comment by nathansobo** [comment]
@bsr203 Do you remember if you saw an exception with the message `Uncaught TypeError: Cannot read property 'destroy' of undefined` as well when you were editing?

**Comment by bsr203** [comment]
most likely. there was indeed one another error which was already reported https://github.com/atom/atom/issues/6899 . And could be the one you mentioned also. sorry, doesn't recollect well the message.

**Comment by nathansobo** [comment]
This is fixed in 1.14.1, which I just published.
